apps/ejabberd/src/mongoose_api_common.erl

```diff
@@ -142,17 +142,26 @@ handle_result(Verb, ok, Req, State) ->
 %%    {halt, Req2, State};
 handle_result(<<"GET">>, {ok, Result}, Req, State) ->
     {jiffy:encode(Result), Req, State};
-handle_result(<<"POST">>, {ok, Res}, Req, State) ->
+handle_result(<<"POST">>, {ok, nocontent}, Req, State) ->
```

Comment:
Looks like this may come from `handle_result(Verb, ok, Req, State)`. The `nocontent` variant is most probably here to handle case when we want to reply `204` to a POST or PUT request.